src/chrome/logging/json.ts

```diff
@@ -0,0 +1,17 @@
+export class JSONObject {
+    private constructor(jsonProperties: object) {
+        Object.assign(this, jsonProperties);
+    }
+
+    public static create<T>(jsonProperties: object): JSONObject & T {
+        return <JSONObject & T>new JSONObject(jsonProperties);
```

Comment:
Why do you need a static constructor that just calls `new`? Can the generic parameter go on the class?

Reply:
We need the type of the Wrapped JSON object to be the same as the unwrapped type for things to type properly. "class JSONObject<T> implements T" is not valid. I could try to modify the return value of the constructor, which I think it's possible, but it's not "nice". So I opted to create a  static method with the whole purpose of retaining the T type of the JSON object.

I've just updated the types to reflect this more clearly

Reply:
I see thanks